src/main/java/org/folio/des/service/bursarlegacy/BursarMigrationService.java

```diff
@@ -0,0 +1,307 @@
+package org.folio.des.service.bursarlegacy;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.des.domain.dto.BursarExportDataToken;
+import org.folio.des.domain.dto.BursarExportFilter;
+import org.folio.des.domain.dto.BursarExportFilterAge;
+import org.folio.des.domain.dto.BursarExportFilterCondition;
+import org.folio.des.domain.dto.BursarExportFilterFeeFineOwner;
+import org.folio.des.domain.dto.BursarExportFilterFeeType;
+import org.folio.des.domain.dto.BursarExportFilterPatronGroup;
+import org.folio.des.domain.dto.BursarExportHeaderFooter;
+import org.folio.des.domain.dto.BursarExportJob;
+import org.folio.des.domain.dto.BursarExportTokenConditional;
+import org.folio.des.domain.dto.BursarExportTokenConditionalConditionsInner;
+import org.folio.des.domain.dto.BursarExportTokenConstant;
+import org.folio.des.domain.dto.BursarExportTokenDateType;
+import org.folio.des.domain.dto.BursarExportTokenFeeAmount;
+import org.folio.des.domain.dto.BursarExportTokenFeeDate;
+import org.folio.des.domain.dto.BursarExportTokenFeeMetadata;
+import org.folio.des.domain.dto.BursarExportTokenLengthControl;
+import org.folio.des.domain.dto.BursarExportTokenUserDataOptional;
+import org.folio.des.domain.dto.BursarExportTransferCriteria;
+import org.folio.des.domain.dto.BursarExportTransferCriteriaElse;
+import org.folio.des.domain.dto.ExportConfig;
+import org.folio.des.domain.dto.ExportConfigWithLegacyBursar;
+import org.folio.des.domain.dto.ExportTypeSpecificParameters;
+import org.folio.des.domain.dto.Job;
+import org.folio.des.domain.dto.JobWithLegacyBursarParameters;
+import org.folio.des.domain.dto.LegacyBursarFeeFines;
+import org.folio.des.domain.dto.LegacyBursarFeeFinesTypeMapping;
+import org.folio.des.domain.dto.LegacyBursarFeeFinesTypeMappings;
+import org.folio.des.service.JobService;
+import org.folio.des.service.config.impl.BurSarFeesFinesExportConfigService;
+import org.folio.des.service.util.JobMapperUtil;
+import org.springframework.stereotype.Service;
+
+@Log4j2
+@Service
+@RequiredArgsConstructor
+public class BursarMigrationService {
+
+  private static final String BURSAR_EXPORT_MIGRATION_HEADER_CONSTANT = "LIB02";
+
+  // this ensures the currently scheduled configuration is updated and new jobs are created with the new schema
+  public void updateLegacyBursarConfigs(BurSarFeesFinesExportConfigService configService) {
```

Comment:
BurSar?

Reply:
That's the old name of that existing class :(
I figured it'd be noisy to change it, so I opted to leave it alone

Reply:
Renamed